doc/proposals/shared-memory-transport/eprosima/interprocess_shared_mem.md

```diff
@@ -0,0 +1,95 @@
+
+The first message the participants send is the multicast discovery message: "I'm here, and I am listening on portN". So they alloc a buffer in its local segment, write the message to that buffer and push the buffer's descriptor through the port0. Observe, in the global segment, how port0's ring-buffer store the descriptors to Data1a(P1), Data2a(P2), Data3a(P3). 
+
+After the discovery phase, participants know the other participants and its "unicast" ports so they can push messages to specific participants.
+
+Finally, let's observe how P1 share Data1b with P3 and Data1c with P2 and P3, this is done by pushing the buffer's descriptors to the corresponding ports, the descriptors are stored in the port's ring-buffer and listeners registered in these ports are notified.
+
+### Design considerations
+
+* **Minimize global interprocess locks**: Interprocess locks are dangerous because, in the case of one of the involved processes crash or hung while holding an interprocess lock, all the collaborating processes could be affected. In this design, the global segment is the critical area, most of accesses will be lock-free reading operations. Interprocess-locks will only be required when registering / unregistering a new listener in a port. Once the setup has been established, the risk of a process locking the whole port is minimized.
+
+* **Scalable number of processes**: Every process creates its own shared memory segments to store the locally generated messages. This is more scalable than having a global segment shared between all the involved processes. 
+
+* **Per application / process customizable memory usage**: Again, a local shared memory segment allows to adapt the size of the segments to the requirements of the application. Imagine for example an application sending two types of data: Video and status info. It could create a big segment to store video frames, and a small segment for the status messages.
+
+### Future improvements
+
+* **Fault tolerance**: As stated in the design considerations, the possibility of a process crashing holding interprocess resources is real. Implementing fault tolerance for these cases is not an easy task. Timeouts, keep alive checks and garbage collectors are some of the things that could be added to the initial design in order to achieve fault tolerance. This will be considered in future revisions.
+
+### Mapping the design to FastRTPS transport layer
+
+* **Locators**: LOCATOR_KIND_SHMEM (16) is defined to identify shared memory transport endpoints. Only the locator's port will be used to match the shared memory port_id.
+
+* **SharedMemTransportDescriptor**: The following values can be customized:
+    * segment_size: Size of the shared memory segment reserved by the transport.
+    * port_queue_capacity: Size, in number of messages, of the shared memory port message queue.
+    * port_overflow_policy: WAIT_ON_OVERFLOW or FAIL_ON_OVERFLOW.
```

Comment:
Do we have default values for these?

Reply:
enum class OverflowPolicy { DISCARD, FAIL };